src/worker.cpp

```diff
@@ -289,53 +281,55 @@ void Worker::getVpdDataInVector(const std::string& vpdFilePath,
             break;
         }
     }
-}
 
-void Worker::fillVPDMap(const std::string& vpdFilePath,
-                        types::VPDMapVariant& vpdMap)
-{
-    logging::logMessage(std::string("Parsing file = ") + vpdFilePath);
-
-    if (vpdFilePath.empty())
+    try
     {
-        throw std::runtime_error("Invalid file path passed to fillVPDMap API.");
+        std::shared_ptr<ParserInterface> parser =
+            ParserFactory::getParser(vpdVector, vpdFilePath, vpdStartOffset);
+        vpdMap = parser->parse();
     }
-
-    if (!std::filesystem::exists(vpdFilePath))
+    catch (const std::exception& ex)
     {
-        throw std::runtime_error("Can't Find physical file");
-    }
+        if (typeid(ex) == std::type_index(typeid(DataException)))
+        {
+            // TODO: Do what needs to be done in case of Data exception.
+            // Uncomment when PEL implementation goes in.
+            /* string errorMsg =
+                 "VPD file is either empty or invalid. Parser failed for [";
+             errorMsg += m_vpdFilePath;
+             errorMsg += "], with error = " + std::string(ex.what());
 
-    size_t vpdStartOffset = 0;
-    std::string fruInventoryPath;
+             additionalData.emplace("DESCRIPTION", errorMsg);
+             additionalData.emplace("CALLOUT_INVENTORY_PATH",
+                                    INVENTORY_PATH + baseFruInventoryPath);
+             createPEL(additionalData, pelSeverity, errIntfForInvalidVPD,
+             nullptr);*/
 
-    // check if offset present.
-    if (m_parsedJson["frus"].contains(vpdFilePath))
-    {
-        for (const auto& item : m_parsedJson["frus"][vpdFilePath])
-        {
-            if (item.find("offset") != item.end())
-            {
-                vpdStartOffset = item["offset"];
-            }
+            // throw generic error from here to inform main caller about
+            // failure.
+            logging::logMessage(ex.what());
+            throw std::runtime_error("Data Exception occurred");
```

Comment:
can you add details to the log message on what vpdfilepath, and may be errno to be specific